lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -0,0 +1,213 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Imports.
+ */
+/** @jsx jsx */
+import { jsx } from "@emotion/react";
+import * as React from "react";
+import { PureComponent, Fragment } from "react";
+import SignInUpTheme from "../../themes/signInUp";
+import FeatureWrapper from "../../../../../components/featureWrapper";
+import { StyleProvider } from "../../../../../styles/styleContext";
+import { defaultPalette } from "../../../../../styles/styles";
+import { getQueryParams, getRedirectToPathFromURL } from "../../../../../utils";
+import Recipe from "../../../recipe";
+import { getStyles } from "../../../components/themes/styles";
+import { RecipeInterface, LoginAttemptInfo } from "../../../types";
+import { ComponentOverrideContext } from "../../../../../components/componentOverride/componentOverrideContext";
+import { FeatureBaseProps } from "../../../../../types";
+import { formatPhoneNumberIntl } from "react-phone-number-input/min";
+
+type SignInUpState = {
+    error: string | undefined;
+    loaded: boolean;
+    loginAttemptInfo: LoginAttemptInfo | undefined;
+};
+
+type PropType = FeatureBaseProps & {
+    recipe: Recipe;
+};
+
+class SignInUp extends PureComponent<PropType, SignInUpState> {
```

Comment:
One missing aspect is that if someone has enabled both userinput and link, and they have redeemed the link code on another tab, then this tab should change to show the "success UI" which shows that close tab cause you have already signed in text.

Reply:
That's one of the TODOs, but I assumed we would just redirect the same as a successful login would. Should I add a separate screen for this? Maybe we could add an extra field to the redirection context?

Reply:
We have a separate screen for this. So please implement that.